### PR TITLE
Gke monitoring

### PR DESCRIPTION
**Pull request by wajihyassine**
The following PR adds Turbinia Prometheus/Grafana deployment scripts for GKE as well as updates the instructions on how to setup monitoring for GKE.
* monitoring/k8s/grafana/turbinia-application-metrics.yaml - Turbinia Application Metrics dashboard configmap for GKE
* monitoring/k8s/turbinia-healthcheck-metrics.yaml - Turbinia Healthcheck Metrics dashboard configmap for GKE
* monitoring/k8s/prometheus/prometheus-additional.yaml - Used for GKE autodiscovery for Turbinia metrics
* monitoring/k8s/prometheus/turbinia-custom-rules.yaml - Imports the Turbinia custom rules to calculate run times.
* docs/user/install-gke.md - updated instructions for setting up GKE monitoring

### CODE REVIEW

File: docs/user/install-gke.md
Comment (aarontp) [inline docs/user/install-gke.md:120]: Maybe make this it's own step that comes before the "test that the changes were properly made..." section?

**Review by aarontp — APPROVED** [review]
LGTM, but I had a question about the .yaml files.  I'll also wait for a review by @hacktobeer before merging this.

A few of these .yaml files look like they are the same or similar to the existing .json files that are checked in elsewhere.  Will we keep both sets of files for the long term?  How were these created?  If they were generated, would it make sense to generate them from the .json files on deployment instead of keeping both copies checked in?  Otherwise if we want to change the file, it might be more difficult to try to manually edit different versions of these files and it increases the chance that they'll get out of sync.

File: docs/user/install-gke.md
Comment (wajihyassine) [reply docs/user/install-gke.md:122]: Sounds good, added a section named **Testing Prometheus Deployment** before Grafana setup to seperate it out more.

File: monitoring/k8s/grafana/turbinia-application-metrics.yaml
Comment (aarontp) [inline monitoring/k8s/grafana/turbinia-application-metrics.yaml:1]: This file looks like it's only about 10 lines of header/footer data, and the rest is the same as the corresponding .json file.   Just to reduce the amount of duplicate data, and to make it easier to keep things in sync, could we make this a template file, and then just run a sed command to include the other file?

One way to do it with sed is to have the template file:
```
apiVersion: v1
items:
- apiVersion: v1
  data:
      turbinia-application-metrics.json: |-
@@JSONDATA@@
  kind: ConfigMap
  metadata:
    labels:
      app.kubernetes.io/component: grafana
      app.kubernetes.io/name: grafana
      app.kubernetes.io/part-of: kube-prometheus
      app.kubernetes.io/version: 7.5.4
    name: turbinia-application-metrics
    namespace: monitoring
kind: ConfigMapList
```

And then a one-liner to replace the @@JSONDATA@@ with the other file:
```
sed -e '/@@JSONDATA@@/{r turbinia-application-metrics.json' -e 'd}' -i turbinia-application-metrics.yaml
```

Since there are a few files, we could also put the sed lines into short `gen-yaml.sh` script or something.  Then later when we have the updated version of `deploy.sh` we can just include those there.

Comment (hacktobeer) [reply monitoring/k8s/grafana/turbinia-application-metrics.yaml:1]: Agree with that approach.

**Review by hacktobeer — APPROVED** [review]
LGTM if Aarons comments are resolved.

File: monitoring/k8s/grafana/turbinia-application-metrics.yaml
Comment (wajihyassine) [reply monitoring/k8s/grafana/turbinia-application-metrics.yaml:1]: Added a script named `gen-yaml.sh` that automatically updates the application and healthcheck dashboards and the prometheus rules section

**Review by aarontp — APPROVED** [review]
LGTM!

**Pull request merged**
